In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/gufhtugu-publications-dataset-challenge/GP Orders - 2.csv
/kaggle/input/gufhtugu-publications-dataset-challenge/GP Orders - 4.csv
/kaggle/input/gufhtugu-publications-dataset-challenge/GP Orders - 5.csv


In [5]:
df2 = pd.read_csv('../input/gufhtugu-publications-dataset-challenge/GP Orders - 2.csv', encoding='ISO-8859-1')
df4 = pd.read_csv('../input/gufhtugu-publications-dataset-challenge/GP Orders - 4.csv', encoding='ISO-8859-1')
df5 = pd.read_csv('../input/gufhtugu-publications-dataset-challenge/GP Orders - 5.csv', encoding='ISO-8859-1')

In [6]:
df2.head()

Order Number Order Status                            Book Name  \
0         70978    Completed           Column Nigari Kese Sekhain   
1         70962    Completed               ??????? ?? ???? ??????   
2         70960    Completed  Justju ka safar-1/Waqfa e Pareshani   
3         70959    Completed                          R ka Taaruf   
4         70958    Completed               ??????? ?? ???? ??????   

        Order Date City (Billing)  
0  1/20/2021 17:43           Swat  
1  1/20/2021 13:37        Karachi  
2  1/20/2021 10:48        Karachi  
3  1/20/2021 10:32        Karachi  
4  1/20/2021 10:29        Karachi

In [7]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19187 entries, 0 to 19186
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Order Number    19187 non-null  int64 
 1   Order Status    19187 non-null  object
 2   Book Name       19185 non-null  object
 3   Order Date      19187 non-null  object
 4   City (Billing)  19186 non-null  object
dtypes: int64(1), object(4)
memory usage: 749.6+ KB


In [10]:
df2.describe()

Order Number
count  19187.000000
mean   52483.815344
std    15826.191039
min      307.000000
25%    48048.500000
50%    56394.000000
75%    63724.500000
max    70978.000000

## Descriptive Statistics of GP Orders - 2

*  `Total rows` = 19187
*  `Columns` = 5
*  `Continuous Variable(s)` = 1
*  `Categorical Variable(s)` = 4
*  `Missing Values` = 1 (City)

count: The total number of order records in the dataset is 19,187.

mean: The mean (average) value of the "Order Number" is approximately 52,483.82. This indicates that the average order number in the dataset is around 52,483.

std: The standard deviation of the "Order Number" is approximately 15,826.19. The standard deviation measures the amount of variation or dispersion in the dataset. A higher standard deviation indicates that the "Order Number" values are more spread out from the mean.

min: The minimum "Order Number" value in the dataset is 307. This is the smallest order number present in the dataset.

25%: The first quartile (25th percentile) of the "Order Number" values is 48,048.5. This means that 25% of the orders have "Order Number" values below this value.

50%: The second quartile (50th percentile), also known as the median, of the "Order Number" values is 56,394. This represents the middle value of the dataset when it is sorted in ascending order. Half of the orders have "Order Number" values below this value, and the other half have values above it.

75%: The third quartile (75th percentile) of the "Order Number" values is 63,724.5. This means that 75% of the orders have "Order Number" values below this value.

max: The maximum "Order Number" value in the dataset is 70,978. This is the largest order number present in the dataset.



In [12]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19187 entries, 0 to 19186
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Order Number  19187 non-null  int64 
 1   Order Status     19187 non-null  object
 2   Book Name        19185 non-null  object
 3   Order Date       19187 non-null  object
 4   City (Billing)   19186 non-null  object
dtypes: int64(1), object(4)
memory usage: 749.6+ KB


In [11]:
df4.describe()

Order Number
count     19187.000000
mean      52483.815344
std       15826.191039
min         307.000000
25%       48048.500000
50%       56394.000000
75%       63724.500000
max       70978.000000

 ## Observation
 'GP - 4' dataset is identical to 'GP - 2' dataset.
 The only difference is about extra missing values in GP - 4 dataset. 
 2 missing values are found in 'Book Name' column. 
 1 missing value in 'City (Billing)' column.

In [13]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19239 entries, 0 to 19238
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Order Number       19239 non-null  int64  
 1   Order Status          19239 non-null  object 
 2   Book Name             19237 non-null  object 
 3   Order Date & Time     19239 non-null  object 
 4   City                  19238 non-null  object 
 5   Payment Method        19229 non-null  object 
 6   Total items           19239 non-null  int64  
 7   Total weight (grams)  19239 non-null  float64
dtypes: float64(1), int64(2), object(5)
memory usage: 1.2+ MB


In [14]:
df5.describe()

Order Number   Total items  Total weight (grams)
count     19239.000000  19239.000000          19239.000000
mean      52535.780238      6.051874            363.749675
std       15835.761148    576.938452            834.097325
min         307.000000      0.000000            200.000000
25%       48064.500000      1.000000            200.000000
50%       56436.000000      1.000000            200.000000
75%       63779.500000      1.000000            500.000000
max       71420.000000  80022.000000          55000.000000